airflow/www/views.py

```diff
@@ -3465,25 +3466,42 @@ def grid_data(self):
         if num_runs is None:
             num_runs = conf.getint("webserver", "default_dag_run_display_number")
 
-        try:
-            base_date = timezone.parse(request.args["base_date"], strict=True)
-        except (KeyError, ValueError):
-            base_date = dag.get_latest_execution_date() or timezone.utcnow()
+        dagrun = None
+        if run_id:
+            with create_session() as session:
+                dagrun = dag.get_dagrun(run_id=run_id, session=session)
+                if not dagrun:
+                    return {"error": f"can't find dag_run_id={run_id}"}, 404
+            base_date = dagrun.execution_date
```

Comment:
The old code in `get_date_time_num_runs_dag_runs_form_data` that does similar lookups has following workaround, is that still needed here also? 

        # The DateTimeField widget truncates milliseconds and would loose
        # the first dag run. Round to next second.
        base_date = (date_time + datetime.timedelta(seconds=1)).replace(microsecond=0)

I didn't see any such issue during my short exploratory testing of this.

Reply:
I think this was specific to the widget so we're ok here.